### ISSUE TITLE
Coq's Proof General fails to install on first run.

### ISSUE THREAD

**Issue by brettgilio** [hlissner/doom-emacs#2788]
**What did you expect to happen?**
Proof General to build and install without failure.

**What actually happened?**
Proof general fails to build and install on first run, but builds and installs fine on subsequent runs.

**Additional details:**
This has been occurring for me for months, but I always forgot to actually run with the debug flag and report it.

<details><pre>
    → Building proof-general...
    → Building proof-general...done

...
  (let* ((plist (append (list :recipe recipe) recipe)) (recipe (plist-get plist :recipe)) (package (plist-get plist :package)) (type (plist-get plist :type)) (local-repo (plist-get plist :local-repo))) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output))))))
  (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output))))))
  (catch (quote --cl-block-nil--) (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output)))))))
  (cl--block-wrapper (catch (quote --cl-block-nil--) (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output))))))))
  (cl-block nil (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output)))))))
  (while --dolist-tail-- (setq recipe (car --dolist-tail--)) (cl-block nil (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output))))))) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (let ((--dolist-tail-- recipes) recipe) (while --dolist-tail-- (setq recipe (car --dolist-tail--)) (cl-block nil (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e ...)))))) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (dolist (recipe recipes) (cl-block nil (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output))))))))
  (let* ((recipes (doom-package-recipe-list)) (built nil) (straight-use-package-pre-build-functions (cons (lambda (pkg) (cl-pushnew pkg built :test (function equal))) straight-use-package-pre-build-functions))) (dolist (recipe recipes) (cl-block nil (straight--with-plist (append (list :recipe recipe) recipe) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e ...))))))) (nreverse built))
  (doom-with-package-recipes (doom-package-recipe-list) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output))))))
  (and t (doom-with-package-recipes (doom-package-recipe-list) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal (quote doom-package-error) (list package e (straight--process-get-output)))))))
  (let* ((built (and t (doom-with-package-recipes (doom-package-recipe-list) (recipe package type local-repo) (condition-case e (straight-use-package (intern package)) ((debug error) (signal ... ...))))))) (if built (doom--format-print (doom--format (format (doom--format-class (quote success) "Installed %d packages") (length built)))) (progn (doom--format-print (doom--format (format (doom--format-class (quote info) "No packages need to be installed")))) nil)))
  (let ((doom-format-indent (+ doom-format-indent-increment doom-format-indent))) (let* ((built (and t (doom-with-package-recipes (doom-package-recipe-list) (recipe package type local-repo) (condition-case e (straight-use-package ...) (... ...)))))) (if built (doom--format-print (doom--format (format (doom--format-class (quote success) "Installed %d packages") (length built)))) (progn (doom--format-print (doom--format (format (doom--format-class ... "No packages need to be installed")))) nil))))
  (let ((pinned (doom-package-pinned-list))) (let ((doom-format-indent (+ doom-format-indent-increment doom-format-indent))) (let* ((built (and t (doom-with-package-recipes (doom-package-recipe-list) (recipe package type local-repo) (condition-case e ... ...))))) (if built (doom--format-print (doom--format (format (doom--format-class ... "Installed %d packages") (length built)))) (progn (doom--format-print (doom--format (format ...))) nil)))))
  doom-cli-packages-install()
  (let ((doom-format-indent (+ doom-format-indent-increment doom-format-indent))) (if (and (not inhibit-envvar-p) (file-exists-p doom-env-file)) (progn (doom-cli-reload-env-file (quote force)))) (doom-cli-reload-core-autoloads) (doom-cli-packages-install) (doom-cli-packages-build) (doom-cli-packages-purge prune-p (quote builds-p) prune-p prune-p) (doom-cli-reload-package-autoloads) t)
  (let (success) (let ((--dolist-tail-- (list doom-autoload-file doom-package-autoload-file))) (while --dolist-tail-- (let ((file (car --dolist-tail--))) (delete-file file) (delete-file (byte-compile-dest-file file)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (doom-initialize (quote force) (quote noerror)) (doom-initialize-modules) (doom--format-print (doom--format (format (doom--format-class (quote start) "Synchronizing your config with Doom Emacs...")))) (let ((doom-format-indent (+ doom-format-indent-increment doom-format-indent))) (if (and (not inhibit-envvar-p) (file-exists-p doom-env-file)) (progn (doom-cli-reload-env-file (quote force)))) (doom-cli-reload-core-autoloads) (doom-cli-packages-install) (doom-cli-packages-build) (doom-cli-packages-purge prune-p (quote builds-p) prune-p prune-p) (doom-cli-reload-package-autoloads) t))
  (let ((if-necessary-p (cdr (assq (quote if-necessary-p) --alist--))) (inhibit-envvar-p (cdr (assq (quote inhibit-envvar-p) --alist--))) (prune-p (cdr (assq (quote prune-p) --alist--)))) :bare t (let (success) (let ((--dolist-tail-- (list doom-autoload-file doom-package-autoload-file))) (while --dolist-tail-- (let ((file (car --dolist-tail--))) (delete-file file) (delete-file (byte-compile-dest-file file)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (doom-initialize (quote force) (quote noerror)) (doom-initialize-modules) (doom--format-print (doom--format (format (doom--format-class (quote start) "Synchronizing your config with Doom Emacs...")))) (let ((doom-format-indent (+ doom-format-indent-increment doom-format-indent))) (if (and (not inhibit-envvar-p) (file-exists-p doom-env-file)) (progn (doom-cli-reload-env-file (quote force)))) (doom-cli-reload-core-autoloads) (doom-cli-packages-install) (doom-cli-packages-build) (doom-cli-packages-purge prune-p (quote builds-p) prune-p prune-p) (doom-cli-reload-package-autoloads) t)))
  (closure ((plist :bare t :group "Maintenance") (aliases s refresh re) (name . sync) cl-struct-doom-cli-option-tags cl-struct-doom-cli-tags t) (--alist--) (ignore --alist--) (let ((if-necessary-p (cdr (assq (quote if-necessary-p) --alist--))) (inhibit-envvar-p (cdr (assq (quote inhibit-envvar-p) --alist--))) (prune-p (cdr (assq (quote prune-p) --alist--)))) :bare t (let (success) (let ((--dolist-tail-- (list doom-autoload-file doom-package-autoload-file))) (while --dolist-tail-- (let ((file ...)) (delete-file file) (delete-file (byte-compile-dest-file file)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (doom-initialize (quote force) (quote noerror)) (doom-initialize-modules) (doom--format-print (doom--format (format (doom--format-class (quote start) "Synchronizing your config with Doom Emacs...")))) (let ((doom-format-indent (+ doom-format-indent-increment doom-format-indent))) (if (and (not inhibit-envvar-p) (file-exists-p doom-env-file)) (progn (doom-cli-reload-env-file (quote force)))) (doom-cli-reload-core-autoloads) (doom-cli-packages-install) (doom-cli-packages-build) (doom-cli-packages-purge prune-p (quote builds-p) prune-p prune-p) (doom-cli-reload-package-autoloads) t))))(nil)
  funcall((closure ((plist :bare t :group "Maintenance") (aliases s refresh re) (name . sync) cl-struct-doom-cli-option-tags cl-struct-doom-cli-tags t) (--alist--) (ignore --alist--) (let ((if-necessary-p (cdr (assq (quote if-necessary-p) --alist--))) (inhibit-envvar-p (cdr (assq (quote inhibit-envvar-p) --alist--))) (prune-p (cdr (assq (quote prune-p) --alist--)))) :bare t (let (success) (let ((--dolist-tail-- (list doom-autoload-file doom-package-autoload-file))) (while --dolist-tail-- (let (...) (delete-file file) (delete-file ...) (setq --dolist-tail-- ...)))) (doom-initialize (quote force) (quote noerror)) (doom-initialize-modules) (doom--format-print (doom--format (format (doom--format-class ... "Synchronizing your config with Doom Emacs...")))) (let ((doom-format-indent (+ doom-format-indent-increment doom-format-indent))) (if (and (not inhibit-envvar-p) (file-exists-p doom-env-file)) (progn (doom-cli-reload-env-file ...))) (doom-cli-reload-core-autoloads) (doom-cli-packages-install) (doom-cli-packages-build) (doom-cli-packages-purge prune-p (quote builds-p) prune-p prune-p) (doom-cli-reload-package-autoloads) t)))) nil)
  (if cli (funcall (progn (or (and (memq (type-of cli) cl-struct-doom-cli-tags) t) (signal (quote wrong-type-argument) (list (quote doom-cli) cli))) (aref cli 7)) (doom--cli-process cli args)) (user-error "Couldn't find any %S command" command))
  (let* ((cli (and t (doom-cli-get command)))) (if cli (funcall (progn (or (and (memq (type-of cli) cl-struct-doom-cli-tags) t) (signal (quote wrong-type-argument) (list (quote doom-cli) cli))) (aref cli 7)) (doom--cli-process cli args)) (user-error "Couldn't find any %S command" command)))
  doom-cli-execute("sync" nil)
  (and (doom-cli-execute command args) (print! (success "Finished! (%.4fs)") (float-time (time-subtract (current-time) start-time))))
  (let ((start-time (current-time))) (and (doom-cli-execute command args) (print! (success "Finished! (%.4fs)") (float-time (time-subtract (current-time) start-time)))))
  (condition-case e (let ((start-time (current-time))) (and (doom-cli-execute command args) (print! (success "Finished! (%.4fs)") (float-time (time-subtract (current-time) start-time))))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" (bold "Type:") (car e)) (print! (bold "Message:")) (print-group! (print! "%s" (get (car e) (quote error-message)))) (print! (bold "Data:")) (print-group! (if (cdr e) (let ((--dolist-tail-- ...) item) (while --dolist-tail-- (setq item ...) (print! "%S" item) (setq --dolist-tail-- ...))) (print! "n/a"))) (if (featurep (quote straight)) (progn (if (string-match-p (regexp-quote straight-process-buffer) (error-message-string e)) (progn (print! ...) (print-group! ...)))))) (if debug-on-error nil (terpri) (print! (concat "Run the command again with the -d (or --debug) switch to enable debug\n" "mode and (hopefully) generate a backtrace from this error:\n" "\n  %s\n\n" "If you file a bug report, please include it!") (string-join (append (list (file-name-nondirectory load-file-name) "-d" command) args) " ")) (error ""))))
  (cond ((not noninteractive) (print! "Doom launched out of %s (test mode)" (path user-emacs-directory)) (load! "init.el" user-emacs-directory) (doom-run-all-startup-hooks-h)) ((null command) (doom-cli-execute "help")) ((condition-case e (let ((start-time (current-time))) (and (doom-cli-execute command args) (print! (success "Finished! (%.4fs)") (float-time (time-subtract ... start-time))))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" (bold "Type:") (car e)) (print! (bold "Message:")) (print-group! (print! "%s" (get ... ...))) (print! (bold "Data:")) (print-group! (if (cdr e) (let ... ...) (print! "n/a"))) (if (featurep (quote straight)) (progn (if ... ...)))) (if debug-on-error nil (terpri) (print! (concat "Run the command again with the -d (or --debug) switch to enable debug\n" "mode and (hopefully) generate a backtrace from this error:\n" "\n  %s\n\n" "If you file a bug report, please include it!") (string-join (append ... args) " ")) (error ""))))))
  (let ((help-p (cdr (assq (quote help-p) --alist--))) (debug-p (cdr (assq (quote debug-p) --alist--))) (yes-p (cdr (assq (quote yes-p) --alist--))) (emacsdir (cdr (assq (quote emacsdir) --alist--))) (doomdir (cdr (assq (quote doomdir) --alist--))) (localdir (cdr (assq (quote localdir) --alist--))) (command (cdr (assq (quote command) --alist--))) (args (cdr (assq (quote args) --alist--)))) :bare t (if emacsdir (progn (setq user-emacs-directory (file-name-as-directory emacsdir)) (print! (info "EMACSDIR=%s") localdir))) (if doomdir (progn (setenv "DOOMDIR" doomdir) (print! (info "DOOMDIR=%s") localdir))) (if localdir (progn (setenv "DOOMLOCALDIR" localdir) (print! (info "DOOMLOCALDIR=%s") localdir))) (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-mode t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args (cons command args)))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (if (or emacsdir doomdir localdir) (progn (load! "core/core.el" user-emacs-directory))) (cond ((not noninteractive) (print! "Doom launched out of %s (test mode)" (path user-emacs-directory)) (load! "init.el" user-emacs-directory) (doom-run-all-startup-hooks-h)) ((null command) (doom-cli-execute "help")) ((condition-case e (let ((start-time (current-time))) (and (doom-cli-execute command args) (print! (success "Finished! (%.4fs)") (float-time ...)))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" (bold "Type:") (car e)) (print! (bold "Message:")) (print-group! (print! "%s" ...)) (print! (bold "Data:")) (print-group! (if ... ... ...)) (if (featurep ...) (progn ...))) (if debug-on-error nil (terpri) (print! (concat "Run the command again with the -d (or --debug) switch to enable debug\n" "mode and (hopefully) generate a backtrace from this error:\n" "\n  %s\n\n" "If you file a bug report, please include it!") (string-join ... " ")) (error "")))))))
  (lambda (--alist--) (ignore --alist--) (let ((help-p (cdr (assq (quote help-p) --alist--))) (debug-p (cdr (assq (quote debug-p) --alist--))) (yes-p (cdr (assq (quote yes-p) --alist--))) (emacsdir (cdr (assq (quote emacsdir) --alist--))) (doomdir (cdr (assq (quote doomdir) --alist--))) (localdir (cdr (assq (quote localdir) --alist--))) (command (cdr (assq (quote command) --alist--))) (args (cdr (assq (quote args) --alist--)))) :bare t (if emacsdir (progn (setq user-emacs-directory (file-name-as-directory emacsdir)) (print! (info "EMACSDIR=%s") localdir))) (if doomdir (progn (setenv "DOOMDIR" doomdir) (print! (info "DOOMDIR=%s") localdir))) (if localdir (progn (setenv "DOOMLOCALDIR" localdir) (print! (info "DOOMLOCALDIR=%s") localdir))) (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-mode t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args (cons command args)))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (if (or emacsdir doomdir localdir) (progn (load! "core/core.el" user-emacs-directory))) (cond ((not noninteractive) (print! "Doom launched out of %s (test mode)" (path user-emacs-directory)) (load! "init.el" user-emacs-directory) (doom-run-all-startup-hooks-h)) ((null command) (doom-cli-execute "help")) ((condition-case e (let ((start-time ...)) (and (doom-cli-execute command args) (print! ... ...))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" ... ...) (print! ...) (print-group! ...) (print! ...) (print-group! ...) (if ... ...)) (if debug-on-error nil (terpri) (print! ... ...) (error ""))))))))(((command . "sync")))
  funcall((lambda (--alist--) (ignore --alist--) (let ((help-p (cdr (assq (quote help-p) --alist--))) (debug-p (cdr (assq (quote debug-p) --alist--))) (yes-p (cdr (assq (quote yes-p) --alist--))) (emacsdir (cdr (assq (quote emacsdir) --alist--))) (doomdir (cdr (assq (quote doomdir) --alist--))) (localdir (cdr (assq (quote localdir) --alist--))) (command (cdr (assq (quote command) --alist--))) (args (cdr (assq (quote args) --alist--)))) :bare t (if emacsdir (progn (setq user-emacs-directory (file-name-as-directory emacsdir)) (print! (info "EMACSDIR=%s") localdir))) (if doomdir (progn (setenv "DOOMDIR" doomdir) (print! (info "DOOMDIR=%s") localdir))) (if localdir (progn (setenv "DOOMLOCALDIR" localdir) (print! (info "DOOMLOCALDIR=%s") localdir))) (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-mode t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args ...))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (if (or emacsdir doomdir localdir) (progn (load! "core/core.el" user-emacs-directory))) (cond ((not noninteractive) (print! "Doom launched out of %s (test mode)" (path user-emacs-directory)) (load! "init.el" user-emacs-directory) (doom-run-all-startup-hooks-h)) ((null command) (doom-cli-execute "help")) ((condition-case e (let (...) (and ... ...)) (user-error (print! ... ...) (print! ... ...) (error "")) ((debug error) (print! ...) (print-group! ... ... ... ... ... ...) (if debug-on-error nil ... ... ...))))))) ((command . "sync")))
  (if cli (funcall (progn (or (and (memq (type-of cli) cl-struct-doom-cli-tags) t) (signal (quote wrong-type-argument) (list (quote doom-cli) cli))) (aref cli 7)) (doom--cli-process cli args)) (user-error "Couldn't find any %S command" command))
  (let* ((cli (and t (doom-cli-get command)))) (if cli (funcall (progn (or (and (memq (type-of cli) cl-struct-doom-cli-tags) t) (signal (quote wrong-type-argument) (list (quote doom-cli) cli))) (aref cli 7)) (doom--cli-process cli args)) (user-error "Couldn't find any %S command" command)))
  doom-cli-execute(:main ("sync"))
  (let* ((load-prefer-newer t) (loaddir (file-name-directory (file-truename load-file-name))) (emacsdir (getenv "EMACSDIR")) (user-emacs-directory (abbreviate-file-name (or emacsdir (expand-file-name "../" loaddir))))) (setq load-path (cons (expand-file-name "core" user-emacs-directory) load-path)) (require (quote core)) (require (quote core-cli)) (defcli! :main ((help-p ["-h" "--help"] "Same as help command") (debug-p ["-d" "--debug"] "Turns on doom-debug-mode (and debug-on-error)") (yes-p ["-y" "--yes"] "Auto-accept all confirmation prompts") (emacsdir ["--emacsdir" dir] "Use the emacs config at DIR (e.g. ~/.emacs.d)") (doomdir ["--doomdir" dir] "Use the private module at DIR (e.g. ~/.doom.d)") (localdir ["--localdir" dir] "Use DIR as your local storage directory") &optional command &rest args) "A command line interface for managing Doom Emacs.\n\nIncludes package management, diagnostics, unit tests, and byte-compilation.\n\nThis tool also makes it trivial to launch Emacs out of a different folder or\nwith a different private module." :bare t (if emacsdir (progn (setq user-emacs-directory (file-name-as-directory emacsdir)) (print! (info "EMACSDIR=%s") localdir))) (if doomdir (progn (setenv "DOOMDIR" doomdir) (print! (info "DOOMDIR=%s") localdir))) (if localdir (progn (setenv "DOOMLOCALDIR" localdir) (print! (info "DOOMLOCALDIR=%s") localdir))) (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-mode t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args (cons command args)))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (if (or emacsdir doomdir localdir) (progn (load! "core/core.el" user-emacs-directory))) (cond ((not noninteractive) (print! "Doom launched out of %s (test mode)" (path user-emacs-directory)) (load! "init.el" user-emacs-directory) (doom-run-all-startup-hooks-h)) ((null command) (doom-cli-execute "help")) ((condition-case e (let ((start-time ...)) (and (doom-cli-execute command args) (print! ... ...))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" ... ...) (print! ...) (print-group! ...) (print! ...) (print-group! ...) (if ... ...)) (if debug-on-error nil (terpri) (print! ... ...) (error ""))))))) (doom-cli-execute :main (cdr (member "--" argv))) (setq argv nil))
  eval-buffer(#<buffer  *load*> nil "/home/brettg/.emacs.d/bin/doom" nil t)  ; Reading at buffer position 5425
  load-with-code-conversion("/home/brettg/.emacs.d/bin/doom" "/home/brettg/.emacs.d/bin/doom" nil t)
  load("/home/brettg/.emacs.d/bin/doom" nil t t)
  command-line-1(("-scriptload" "./.emacs.d/bin/doom" "--" "sync"))
  command-line()
  normal-top-level()

</pre></details>

**Steps to reproduce:**
1. Enable coq
2. `doom sync`

**System information:**
<details><pre>
((emacs
  (version . "26.3")
  (features . "XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2")
  (build . "Aug 29, 2019")
  (buildopts "--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now")
  (windowsys . batch)
  (daemonp))
 (doom
  (version . "2.0.9")
  (build . "HEAD -> develop, origin/develop, origin/HEAD 064603f6e 2020-03-28 20:08:53 -0400")
  (dir . "~/.doom.d/"))
 (system
  (type . gnu/linux)
  (config . "x86_64-pc-linux-gnu")
  (shell . "/bin/bash")
  (uname . "Linux 5.4.28-1-MANJARO #1 SMP PREEMPT Wed Mar 25 12:45:29 UTC 2020 x86_64")
  (path "~/.local/bin" "/usr/local/bin" "/usr/local/sbin" "/usr/bin" "/usr/bin/site_perl" "/usr/bin/vendor_perl" "/usr/bin/core_perl" "/usr/lib/emacs/26.3/x86_64-pc-linux-gnu"))
 (config
  (envfile . envvar-file)
  (elc-files . 0)
  (modules :completion company ivy :ui doom fill-column hl-todo indent-guides modeline neotree ophints (popup +defaults) vc-gutter vi-tilde-fringe window-select workspaces :editor file-templates fold snippets :emacs dired electric vc :term eshell shell term vterm :checkers syntax :tools debugger (eval +overlay) lookup lsp magit make pdf :lang assembly (cc +lsp) common-lisp coq data emacs-lisp fstar latex markdown ocaml (org +dragndrop +present) scheme sh :config (default +bindings +smartparens))
  (packages (sml-mode))
  (unpin "n/a")
  (elpa "n/a")))

</pre></details>

**Comment by sorawee** [comment]
I believe I have the same issue. Moreover, `doom build` produces the following error:

```
  → Building proof-general...
  → Building proof-general...done
x There was an unexpected error:
  Type: file-missing
  Message:
    File is missing
  Data:
    "Cannot open load file"
    "No such file or directory"
    "/Users/sorawee/.emacs.d/bin/generic/proof-site"
```

@brettgilio can you try `doom build` and see if you get the same output?

**Comment by t-e-r-m** [comment]
A workaround for this is linking the `PG/generic` folder to `.emacs.d/bin`.

```shell
$ cd ~/..emacs.d/bin
$ ln -s ../.local/straight/repos/PG/generic .
```

I still find it easier to not build `Proof-General` while updating/upgrading. I add it in later and then build it.

With `native-comp`, stalled installs lead to uncompiled files (I think).

**Comment by zeeshanlakhani** [comment]
I had this issue in moving from emacs-plus@26 to emacs-plus@28 particularly.

**Comment by NightMachinary** [comment]
I have this issue as well:

```
  → Building proof-general...
  → Building proof-general...done
x There was an unexpected error:
  Type: file-missing
  Message:
    File is missing
  Data:
    "Cannot open load file"
    "No such file or directory"
    "/Users/evar/.emacs.d.doom/bin/generic/proof-site"
```

**Comment by brabalan** [comment]
I have the same issue:
```
  → Building proof-general...
  → Building proof-general...done
x There was an unexpected error:
  Type: file-missing
  Message:
    File is missing
  Data:
    "Cannot open load file"
    "No such file or directory"
    "/home/schmitta/.emacs.d/bin/generic/proof-site"
```

**Comment by hlissner** [comment]
Is this still an issue?

**Comment by brabalan** [comment]
Yes. My current workaround is to have a symbolic link `bin/generic -> ../.local/straight/repos/PG/generic` for building to succeed.

**Comment by brettgilio** [comment]
Yes, this issue is still present.

**Comment by rgrinberg** [comment]
This issue is still present. Here's the full stack trace:

```
    → Building proof-general...done
    x There was an unexpected error
      Message: File is missing
      Data: (file-missing . "Cannot open load file")
      Backtrace:
        (require proof-site "/Users/rgrinberg/.emacs.d/bin/generic/proof-site")
        (if (and (boundp 'byte-compile-current-file) byte-compile-current-file) ni
        (eval-buffer #<buffer  *load*-865481> nil "/Users/rgrinberg/.emacs.d/.loca
        (load-with-code-conversion "/Users/rgrinberg/.emacs.d/.local/straight/buil
        (load "/Users/rgrinberg/.emacs.d/.local/straight/build-27.1/proof-general/
        (progn (load autoloads-file nil 'nomessage))
        (if (file-exists-p autoloads-file) (progn (load autoloads-file nil 'nomess
        (let ((autoloads-file (straight--autoloads-file package))) (if (file-exist
        (straight--load-package-autoloads "proof-general")
        (if straight-cache-autoloads (progn (if (straight--checkhash package strai
```

**Comment by cpitclaudel** [comment]
Maybe the following helps understand what's going on? The error comes from this code:

```elisp
;;;###autoload
(eval-and-compile
  (defvar pg-init--script-full-path
    (or (and load-in-progress load-file-name)
        (bound-and-true-p byte-compile-current-file)
        (buffer-file-name)))
  (defvar pg-init--pg-root
    (file-name-directory pg-init--script-full-path)))

;;;###autoload
(unless (bound-and-true-p byte-compile-current-file)
  ;; This require breaks compilation, so it must only run when loading the package.
  (require 'proof-site (expand-file-name "generic/proof-site" pg-init--pg-root)))
```

This snippet figures out where the current file is (`pg-init--script-full-path`) and loads Proof General's custom init file (essentially a bunch of custom autoloads) . 

It looks like `load-file-name` is in `/Users/rgrinberg/.emacs.d/bin/`, which is odd.  Is that due to the way Doom loads things?

**Comment by brettgilio** [comment]
> It looks like `load-file-name` is in `/Users/rgrinberg/.emacs.d/bin/`, which is odd. Is that due to the way Doom loads things?

Yeah, it's because the package synchronizer for Doom is being ran out of that directory.

**Comment by cpitclaudel** [comment]
That's still odd, isn't it? (PG isn't using current-directory, it's using load-file-name)

**Comment by brettgilio** [comment]
> Yes. My current workaround is to have a symbolic link `bin/generic -> ../.local/straight/repos/PG/generic` for building to succeed.

@cpitclaudel If I had to make an inference, I would say that generic/proof-site is being expected to exist wherever the bytecompiler is being called from. I'd have to play with the edebugger to be sure, but I think load-file-name is probably being overridden by Doom? Im not entirely sure.

TLDR. I think (read: would guess) some in-built behavior of how bin/doom is written is probably the issue.

**Comment by rgrinberg** [comment]
> and loads Proof General's custom init file (essentially a bunch of custom autoloads) .

Just curious, why does pg need a custom init file? Can't it just use autoload cookies like other packages?

**Comment by brettgilio** [comment]
> > and loads Proof General's custom init file (essentially a bunch of custom autoloads) .
> 
> Just curious, why does pg need a custom init file? Can't it just use autoload cookies like other packages?

PG is huge. I think a lot of people like having a customizable experience with what is and isnt loaded, rather than relying on autoloads for most things.

That said, I'd prefer the autoloads.

**Comment by cpitclaudel** [comment]
> Just curious, why does pg need a custom init file? Can't it just use autoload cookies like other packages?

Something with autoloads in subfolders, IIRC.  I did the refactorign to make PG a proper package a while ago and the long story short is that it has a fairly complex setup and the least disruptive way to make it work was to make an autoload that loaded `proof-site.el`, which is basically what proof-general.el does.

**Comment by rgrinberg** [comment]
@cpitclaudel This patch seems to fix the issue for me:

```
diff --git a/proof-general.el b/proof-general.el
index d0955088..6f01807c 100644
--- a/proof-general.el
+++ b/proof-general.el
@@ -56,8 +56,8 @@
 ;;;###autoload
 (eval-and-compile
   (defvar pg-init--script-full-path
-    (or (and load-in-progress load-file-name)
-        (bound-and-true-p byte-compile-current-file)
+    (or (bound-and-true-p byte-compile-current-file)
+        (and load-in-progress load-file-name)
         (buffer-file-name)))
   (defvar pg-init--pg-root
     (file-name-directory pg-init--script-full-path)))
```

Seems like if you're byte compiling the current file, that should be preferred over whatever is being loaded.

**Comment by monnier** [comment]
I think swapping the order is just going to push the problem elsewhere.  Having this be run at 3 different times (when loading the autoloads, when compiling proof-general.el, and when loading proof-general.el) is asking for trouble anyway.  The rest of the file is similarly full of brittle assumptions (like "we presume that this file will be compiled before any of the files ...").  I think the only assumption that should be made is that `proof-autoloads.el` will be loaded before loading or compiling any of the other files and the rest should be handled via `require`, so that all the other files can be loaded or compiled in any order.

**Comment by cpitclaudel** [comment]
Yes, proof-site is a mess.  I didn't have the motivation to rewrite it at the time.

**Comment by rgrinberg** [comment]
This should be fixed once PG is updated in doom

**Comment by brabalan** [comment]
I no longer have the symbolic link and I’m no longer seeing the error. I think this can be closed.
